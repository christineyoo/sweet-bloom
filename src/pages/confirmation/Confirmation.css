#confirmation {
  padding: 5rem 10rem;
}

.confirmation-flex {
  display: flex;
  flex-direction: row;
}

.confirmation-flex-1 {
  flex: 1;
  margin-right: 20px;
  padding: 2rem;
  border: 1px solid purple;
  margin-bottom: 1rem;
}

#confirmation-items-flex {
  display: flex;
  flex-direction: row;
}

.confirmation-items-flex-1 {
  flex: 1;
  text-align: left;
}

hr {
  margin: 0 1rem;
}
