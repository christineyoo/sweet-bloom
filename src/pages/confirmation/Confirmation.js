import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import './Confirmation.css';

class Confirmation extends Component {
  totalPrice = () => {
    const itemsInCart = this.props.itemsInCart || [];
    if (itemsInCart.length === 0) return null;
    const itemTotals = itemsInCart.map((item) => +item.price * +item.quantity);
    return itemTotals.reduce((res, curr) => res + curr);
  };

  renderItemSummary = () => {
    const itemsInCart = this.props.itemsInCart || [];
    return itemsInCart.map((item, i) => {
      return (
        <li key={i}>
          {item.name} ({item.quantity})
        </li>
      );
    });
  };

  renderShipAddDetails = () => {
    const { name, address, city, state, zipcode } =
      this.props.shippingData || [];
    return (
      <>
        <p>{name}</p>
        <p>{address}</p>
        <p>
          {city}, {state}, {zipcode}
        </p>
      </>
    );
  };

  renderDeliveryDay = () => {
    if (+this.props.shipping === 10) {
      return <p>Arriving today to</p>;
    } else {
      return <p>Arriving tomorrow to</p>;
    }
  };

  componentWillUnmount() {
    this.props.handleClearCart();
    window.location.reload();
  }

  render() {
    return (
      <div id='confirmation'>
        <article id='confirmation-thank-you'>
          <h1>THANK YOU</h1>
          <h2>ORDER CONFIRMED</h2>
        </article>
        <h3>Order Confirmation</h3>
        <h2>
          Order #{Math.floor(Math.random() * 1000)}-
          {Math.floor(Math.random() * 1000)}-{Math.floor(Math.random() * 1000)}
        </h2>
        <h3>Your Purchased Items</h3>
        <ul>{this.renderItemSummary()}</ul>
        <div className='confirmation-flex'>
          <div className='confirmation-flex-1'>
            <h4>Payment Details</h4>
            <div id='confirmation-items-flex'>
              <p className='confirmation-items-flex-1'>Total before shipping</p>
              <p className='confirmation-items-flex-1'>${this.totalPrice()}</p>
            </div>
            <div id='confirmation-items-flex'>
              <p className='confirmation-items-flex-1'>Shipping</p>
              <p className='confirmation-items-flex-1'>
                ${this.props.shipping}
              </p>
            </div>
            <hr />
            <div id='confirmation-items-flex'>
              <p className='confirmation-items-flex-1'>Order Total</p>
              <p className='confirmation-items-flex-1'>
                ${+this.totalPrice() + +this.props.shipping}
              </p>
            </div>
          </div>
          <div className='confirmation-flex-1'>
            <h4>Delivery Details</h4>
            {this.renderDeliveryDay()}
            {this.renderShipAddDetails()}
          </div>
        </div>
        <Link to='/shop'>
          <button id='landing-button'>Shop more</button>
        </Link>
      </div>
    );
  }
}

export default withRouter(Confirmation);
