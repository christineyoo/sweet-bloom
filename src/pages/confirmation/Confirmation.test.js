import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Confirmation from './Confirmation';

it('renders Confirmation UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <Confirmation />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
