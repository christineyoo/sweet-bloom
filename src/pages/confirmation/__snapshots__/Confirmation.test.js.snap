// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Confirmation UI as expected 1`] = `
<div
  id="confirmation"
>
  <h1>
    Thank you for shopping with Sweet Bloom!
  </h1>
  <h2>
    Your order details are shown below.
  </h2>
  <h3>
    Order Confirmation
  </h3>
  <p>
    Order #123-456-789
  </p>
  <p>
    Items
  </p>
  <ul />
  <div
    className="confirmation-flex"
  >
    <div
      className="confirmation-flex-1"
    >
      <div
        id="confirmation-items-flex"
      >
        <p
          className="confirmation-items-flex-1"
        >
          Total before shipping
        </p>
        <p
          className="confirmation-items-flex-1"
        >
          $
        </p>
      </div>
      <div
        id="confirmation-items-flex"
      >
        <p
          className="confirmation-items-flex-1"
        >
          Shipping
        </p>
        <p
          className="confirmation-items-flex-1"
        >
          $
        </p>
      </div>
      <hr />
      <div
        id="confirmation-items-flex"
      >
        <p
          className="confirmation-items-flex-1"
        >
          Order Total
        </p>
        <p
          className="confirmation-items-flex-1"
        >
          $
          NaN
        </p>
      </div>
    </div>
    <div
      className="confirmation-flex-1"
    >
      <h4>
        Delivery Details
      </h4>
      <p>
        Arriving tomorrow
      </p>
      <p>
        <strong>
          Ship to:
        </strong>
      </p>
      <p>
        <p />
        <p />
        <p>
          , 
          , 
        </p>
      </p>
    </div>
  </div>
  <a
    href="/shop"
    onClick={[Function]}
  >
    <button>
      Shop more
    </button>
  </a>
</div>
`;
